document-loaders/langchain4j-document-loader-github/src/main/java/dev/langchain4j/data/document/loader/github/GithubDocumentLoader.java

```diff
@@ -0,0 +1,143 @@
+package dev.langchain4j.data.document.loader.github;
+
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.DocumentLoader;
+import dev.langchain4j.data.document.DocumentParser;
+import dev.langchain4j.data.document.source.github.GithubSource;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GitHubBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GithubDocumentLoader {
+
+    private static final Logger logger = LoggerFactory.getLogger(GithubDocumentLoader.class);
+
+    private String apiUrl = "https://api.github.com";
+
+    private final GitHub github;
+
+    public GithubDocumentLoader(String apiUrl, String githubToken, String gitHubTokenOrganization) throws IOException {
+        GitHubBuilder gitHubBuilder = new GitHubBuilder();
+        if (apiUrl != null) {
+            this.apiUrl = apiUrl;
+        }
+        gitHubBuilder.withEndpoint(this.apiUrl);
+        if (githubToken != null) {
+            if (gitHubTokenOrganization == null) {
+                gitHubBuilder.withOAuthToken(githubToken);
+            } else {
+                gitHubBuilder.withOAuthToken(githubToken, gitHubTokenOrganization);
+            }
+        }
+        github = gitHubBuilder.build();
+    }
+
+    public GithubDocumentLoader(GitHub github) {
+        this.github = github;
+    }
+
```

Comment:
Oh yes the token is optional (it's recommended otherwise you'll be rate limited, but it works for simple projects).
I've added the empty constructor.